cmd/admin-handlers.go

```diff
@@ -1768,3 +1773,141 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+	var wg sync.WaitGroup
+
+	switch madmin.InfoType(vars[madmin.INFO]) {
+	case madmin.ETCD:
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoints := globalEtcdClient.Endpoints()
+		endPoints := make([]string, len(etcdEndPoints))
+		errorString := make([]string, len(etcdEndPoints))
+
+		for i, endPoint := range etcdEndPoints {
+			wg.Add(1)
+			go func(i int, endPoint string) {
+				defer wg.Done()
+				endPoints[i] = endPoint
+				if err := checkConnection(endPoint); err != nil {
+					errorString[i] = errEtcdUnreachable.Error()
```

Comment:
we shouldn't mask `err`. We can have `err.Error()`

Reply:
done.